### PR TITLE
Add Support for Bitbucket Server

### PR DESCRIPTION
**Pull request by talarian1**
- [ ] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.

---

### CODE REVIEW

File: commands/scanpullrequests.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	clitool "github.com/urfave/cli/v2"
+)
+
+func ScanPullRequests(c *clitool.Context) error {
+	// Get params and VCS client
+	params, client, err := utils.GetParamsAndClient()
+	if err != nil {
+		return err
+	}
+	// Send usage report
```
Comment (eyalbe4) [inline commands/scanpullrequests.go:20]: I suggested to @sverdlov93 in his PR to make changes to the `Send usage report` flow for all commands, so best to stay in sync with him about this.

File: commands/scanpullrequests.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+	// Do scan pull requests
+	err = scanPullRequests(params, client)
+
+	// Wait for usage report
+	<-usageReportSent
+	return err
+}
+
+// Scan pull requests by as follows:
```
Comment (eyalbe4) [inline commands/scanpullrequests.go:32]: ...requests by as follows: --> ...requests as follows:

File: commands/scanpullrequests.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+// a. Retrive all open pull requests
+// b. Find the ones that should be scanned ( new pr or one with re-scan comment)
```
Comment (eyalbe4) [inline commands/scanpullrequests.go:34]: ...( new pr or one with re-scan comment) --> ...(new PRs or PRs with a 're-scan' comment)

File: commands/scanpullrequests.go [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+					RepoOwner:   params.RepoOwner,
+					Repo:        pr.Source.Repository,
+					BaseBranch:  pr.Source.Name,
+				},
+			}
+			wd, err := downloadRepoToTempDir(client, prScanParams)
+			if err != nil {
+				return err
+			}
+			//clean up
```
Comment (eyalbe4) [inline commands/scanpullrequests.go:64]: //clean up --> // Cleanup

File: commands/scanpullrequests.go [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+	if err != nil {
+		return err
+	}
+	for _, pr := range openPullRequests {
+		shouldScan, err := shouldScanPullRequest(params, client, int(pr.ID))
+		if err != nil {
+			return err
+		}
+		if shouldScan {
+			// download the pull request source ("from") branch
```
Comment (eyalbe4) [inline commands/scanpullrequests.go:48]: download --> Download

File: commands/scanpullrequests.go [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+	if err != nil {
+		return
+	}
+	// Sort the comment according to time created, the newest comment should be the first one.
+	sort.Slice(pullRequestsComments, func(i, j int) bool {
+		return pullRequestsComments[i].Created.After(pullRequestsComments[j].Created)
+	})
+
+	for _, comment := range pullRequestsComments {
+		// if found re-scan request
```
Comment (eyalbe4) [inline commands/scanpullrequests.go:118]: // if found re-scan request
-->
// If this a 're-scan' request comment

File: commands/scanpullrequests.go [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+		if isReScanComment(comment.Content) {
+			return true, nil
+		}
+		// if found frogbot comment and not re-scan request, do not scan this pull request.
```
Comment (eyalbe4) [inline commands/scanpullrequests.go:122]: // if found frogbot comment and not re-scan request, do not scan this pull request.
-->
// if this is a Frogbot 'scan rsults' comment and not 're-scan' request comment, do not scan this pull request.

File: commands/scanpullrequests.go [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+		if isFrogbotComment(comment.Content) {
+			return false, nil
+		}
+	}
+	// A new pull request - should be scanned
```
Comment (eyalbe4) [inline commands/scanpullrequests.go:127]: A new pull request - should be scanned
-->
This is a new pull request and it therefore should be scanned.

File: commands/scanpullrequests.go [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+	return true, nil
+}
+
+func isReScanComment(comment string) bool {
+	return strings.ToLower(strings.TrimSpace(comment)) == utils.RescanRequestComment
+}
+
+func isFrogbotComment(comment string) bool {
```
Comment (eyalbe4) [inline commands/scanpullrequests.go:135]: isFrogbotComment --> isFrogbotResultComment

File: commands/scanpullrequests.go [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+		// if found frogbot comment and not re-scan request, do not scan this pull request.
+		if isFrogbotComment(comment.Content) {
+			return false, nil
+		}
+	}
+	// A new pull request - should be scanned
+	return true, nil
+}
+
+func isReScanComment(comment string) bool {
```
Comment (eyalbe4) [inline commands/scanpullrequests.go:131]: isReScanComment --> isFrogbotRescanComment

File: templates/Jenkins/Jenkinsfile [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+
+pipeline {
+    agent any
+
+    triggers {
+        cron(CRON_SETTINGS)
+    }
+
+    environment {
+        // [Mandatory only for project which npm, pip, pipenv, nuget and dotnet]
```
Comment (eyalbe4) [inline templates/Jenkins/Jenkinsfile:12]: [Mandatory only for project which npm, pip, pipenv, nuget and dotnet]
-->
[Mandatory only for projects which use npm, pip, pipenv, nuget and dotnet to download their dependencies]

This same error exists on the GitLab section in the README, so let's fix it there as well.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
Let's think what the best strategy for testing this new functionality.

**Review by sverdlov93 — APPROVED** [review]

**Review by sverdlov93 — COMMENTED** [review]
LGTM

**Pull request merged**